python/ray/tests/test_failure.py

```diff
@@ -291,6 +292,44 @@ def fail_method(self):
     assert error_message1 in errors[0].error_message
 
 
+def test_failed_actor_init_and_restart(ray_start_regular, error_pubsub):
+    p = error_pubsub
+    error_message1 = "actor constructor failed"
+    error_flag_file_name = "/tmp/TEST_ERROR_FLAG_" + str(
+        random.randint(0, 99999))
+
+    @ray.remote(max_restarts=2)
+    class FailedActor:
+        def __init__(self):
+            # This actor will raise error at first time.
+            # Then after restarting, it will be created successfully
+            if not os.path.isfile(error_flag_file_name):
+                open(error_flag_file_name, "w").close()
+                print("flag not exits, raise error")
+                raise Exception(error_message1)
+
+        def ok(self):
+            return "OK"
+
+    # this time actor will raise error in constructor and restart
+    a = FailedActor.remote()
+    errors = get_error_message(p, 1, ray_constants.TASK_PUSH_ERROR)
+    print(errors)
+    assert len(errors) == 1
+    assert errors[0].type == ray_constants.TASK_PUSH_ERROR
+    assert error_message1 in errors[0].error_message
+
+    # finally after restarting, actor will be alive
+    def wait_until_ok():
+        try:
+            return ray.get(a.ok.remote()) == "OK"
```

Comment:
We can return `ray.get_runtime_context().was_current_actor_reconstructed` in the "ok" method and assert the result.

Reply:
same as commented in java tests.